game-core/src/main/java/games/strategy/triplea/ui/MacOsIntegration.java

```diff
@@ -40,8 +41,16 @@ public static void setOpenFileHandler(final Consumer<File> handler) {
   }
 
   /** Sets the specified quit handler to the application. */
-  public static void setQuitHandler(final Runnable handler) {
+  public static void setQuitHandler(final BooleanSupplier handler) {
     checkNotNull(handler);
-    Desktop.getDesktop().setQuitHandler((quitEvent, quitResponse) -> handler.run());
+    Desktop.getDesktop()
+        .setQuitHandler(
+            (quitEvent, quitResponse) -> {
+              if (handler.getAsBoolean()) {
+                quitResponse.performQuit();
+              } else {
+                quitResponse.cancelQuit();
+              }
+            });
```

Comment:
This reminds me of #5283
Any chance if we're using the files handler the wrong way?
#5253